src/components/DocumentationLayout.vue

```diff
@@ -0,0 +1,285 @@
+              />
+              <transition name="delay-hiding">
+                <Navigator
+                  v-show="sidenavVisibleOnMobile || breakpoint === BreakpointName.large"
+                  :flatChildren="slotProps.flatChildren"
+                  :parent-topic-identifiers="parentTopicIdentifiers"
+                  :technology="slotProps.technology || technology"
+                  :is-fetching="slotProps.isFetching"
+                  :error-fetching="slotProps.errorFetching"
+                  :api-changes="slotProps.apiChanges"
+                  :references="references"
+                  :navigator-references="slotProps.references"
+                  :scrollLockID="scrollLockID"
+                  :render-filter-on-top="breakpoint !== BreakpointName.large"
+                  @close="handleToggleSidenav(breakpoint)"
+                >
+                  <template v-if="enableQuickNavigation" #filter>
+                    <QuickNavigationButton @click.native="openQuickNavigationModal" />
+                  </template>
+                  <template #navigator-head>
+                    <slot name="nav-title" />
+                  </template>
+                </Navigator>
+              </transition>
+            </div>
+          </template>
+        </NavigatorDataProvider>
+      </template>
+      <slot name="content" />
+    </AdjustableSidebarWidth>
+  </div>
+</template>
+
+<script>
+import { PortalTarget } from 'portal-vue';
+import QuickNavigationButton from 'docc-render/components/Navigator/QuickNavigationButton.vue';
+import QuickNavigationModal from 'docc-render/components/Navigator/QuickNavigationModal.vue';
+import AdjustableSidebarWidth from 'docc-render/components/AdjustableSidebarWidth.vue';
+import Navigator from 'docc-render/components/Navigator.vue';
+import onPageLoadScrollToFragment from 'docc-render/mixins/onPageLoadScrollToFragment';
+import { BreakpointName } from 'docc-render/utils/breakpoints';
+import { storage } from 'docc-render/utils/storage';
+import { getSetting } from 'docc-render/utils/theme-settings';
+
+import NavigatorDataProvider from 'theme/components/Navigator/NavigatorDataProvider.vue';
+import DocumentationNav from 'theme/components/DocumentationTopic/DocumentationNav.vue';
+
+const NAVIGATOR_HIDDEN_ON_LARGE_KEY = 'navigator-hidden-large';
+
+export default {
+  name: 'DocumentationLayout',
+  constants: { NAVIGATOR_HIDDEN_ON_LARGE_KEY },
+  components: {
+    Navigator,
+    AdjustableSidebarWidth,
+    NavigatorDataProvider,
+    Nav: DocumentationNav,
+    QuickNavigationButton,
+    QuickNavigationModal,
+    PortalTarget,
+  },
+  mixins: [onPageLoadScrollToFragment],
+  props: {
+    enableNavigator: Boolean,
+    diffAvailability: {
+      type: Object,
+      required: false,
+    },
+    interfaceLanguage: {
+      type: String,
+      required: false,
+    },
+    references: {
+      type: Object,
+      default: () => {},
+    },
+    objcPath: {
```

Comment:
We've already discussed this in person, and it's not a blocker for this PR, but I just want to note this for reference:

I think it would be great if we were able to refactor things even more in the future so that this new layout component doesn't need to know about all these props that aren't really tied to the layout itself.